commons-kafka/src/main/scala/com/avsystem/commons/kafka/SerdeRegistry.scala

```diff
@@ -0,0 +1,27 @@
+package com.avsystem.commons
+package kafka
+
+import com.avsystem.commons.kafka.exceptions.UnsupportedVersionEvent
+
+import scala.collection.mutable
+
+
+class SerdeRegistry[T] extends AbstractDeserializer[T] {
+  private val registry = mutable.Map[Byte, KafkaSerde[_ <: T]]()
+
+  def add[X <: T](version: Byte, serde: KafkaSerde[X])(implicit classTag: ClassTag[X]): SerdeRegistry[T] = {
```

Comment:
`classTag` looks unused.

BTW1 general tip: when a method is "fluent" and returns `this`, it's good to give it a return type `this.type` - this way it will work even if it's moved to a superclass or trait.

BTW2 why not add a convenience method that directly takes a `GenCodec`, e.g.

```scala
def add[X <: T](version: Byte)(implicit codec: GenCodec[T]): this.type = {
  registry += (version -> new KafkaSerde(codec))
  this
}
```